akka-actor-typed/src/main/scala/akka/actor/typed/internal/ExtensionsImpl.scala

```diff
@@ -20,7 +20,7 @@ import akka.util.ccompat.JavaConverters._
  * Actor system extensions registry
  */
 @InternalApi
-private[akka] trait ExtensionsImpl extends Extensions { self: ActorSystem[_] =>
+private[akka] trait ExtensionsImpl extends Extensions { self: ActorSystem[_] with InternalRecipientRef[_] =>
```

Comment:
why is `with InternalRecipientRef[_]` needed here?

Reply:
I see that there is some description in the commit comment but I'm not sure I understand and if there is any alternative? https://github.com/akka/akka/pull/29139/commits/8c2aeca5f02aebb4fece2c2cb3ce24d241ad4eea

Reply:
> why is `with InternalRecipientRef[_]` needed here?

Because without `with InternalRecipientRef[_]` => dotty compile error.
The error is because an `ExtensionsImpl` can't be an `ActorSystem` without also be an `InternalRecipientRef`:
```scala
abstract class ActorSystem[-T] extends ActorRef[T] with Extensions with ClassicActorSystemProvider {
  this: InternalRecipientRef[T] =>
```

Reply:
> I see that there is some description in the commit comment but I'm not sure I understand and if there is any alternative? [8c2aeca](https://github.com/akka/akka/commit/8c2aeca5f02aebb4fece2c2cb3ce24d241ad4eea)

Why we need any alternative?
This change is compatible and it fix dotty compile errors